sql/catalyst/src/main/java/org/apache/spark/sql/sources/v2/writer/SupportsDynamicOverwrite.java

```diff
@@ -25,7 +25,7 @@
  * write does not contain data will remain unchanged.
  * <p>
  * This is provided to implement SQL compatible with Hive table operations but is not recommended.
- * Instead, use the {@link SupportsOverwrite overwrite by filter API} to explicitly replace data.
+ * Instead, use the {@code SupportsOverwrite overwrite by filter API} to explicitly replace data.
```

Comment:
`SupportsOverwrite` should still be available, right?

Reply:
Nevermind, I see that it was not moved because the filter API is in the SQL module. We should make a decision about whether we move the public filter API as well. I would prefer to move it to Catalyst.